connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -363,87 +380,97 @@ public EmbeddedKafkaCluster kafka() {
         return kafkaCluster;
     }
 
-    public int executePut(String url, String body) throws IOException {
-        log.debug("Executing PUT request to URL={}. Payload={}", url, body);
-        HttpURLConnection httpCon = (HttpURLConnection) new URL(url).openConnection();
-        httpCon.setDoOutput(true);
-        httpCon.setRequestProperty("Content-Type", "application/json");
-        httpCon.setRequestMethod("PUT");
-        try (OutputStreamWriter out = new OutputStreamWriter(httpCon.getOutputStream())) {
-            out.write(body);
-        }
-        if (httpCon.getResponseCode() < HttpURLConnection.HTTP_BAD_REQUEST) {
-            try (InputStream is = httpCon.getInputStream()) {
-                log.info("PUT response for URL={} is {}", url, responseToString(is));
-            }
-        } else {
-            try (InputStream is = httpCon.getErrorStream()) {
-                log.info("PUT error response for URL={} is {}", url, responseToString(is));
-            }
-        }
-        return httpCon.getResponseCode();
+    /**
+     * Execute a GET request on the given URL.
+     *
+     * @param url the HTTP endpoint
+     * @return the response to the GET request
+     * @throws ConnectException if execution of the GET request fails
+     */
+    public Response executeGet(String url) {
```

Comment:
Changing the return type from `int` to `Response` would be a breaking change. While this test harness is technically not part of the public API of Connect, we did intend that other projects could use this harness as long as they understand that we might change the API in breaking ways. 

But IMO we should still _try_ to evolve the API in ways that are backward compatible, and so I would suggest adding new methods that return `Response`, having the existing methods delegate to them, and deprecate the old methods.

Reply:
That's an excellent point. I overlooked this possibility. 

I've deprecated all the `execute*` methods. Since there can't be overloading on the return type alone I've added the new methods as `request*`. Hope that works.